src/Parser/ElementalShaman/Modules/Talents/Ascendance.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+import SpellIcon from 'common/SpellIcon';
+import { formatPercentage } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+
+class Ascendance extends Module {
+  numLavaBurstsCast = 0;
+  numLightningBoltsCast = 0;
+  numOtherCasts = 0;
+
+  numCasts = {
+    [SPELLS.LAVA_BURST.id]: 1,
+    [SPELLS.EARTH_SHOCK.id]: 1,
+    [SPELLS.ELEMENTAL_BLAST.id]: 1,
+    [SPELLS.FLAME_SHOCK.id]: 1,
+    [SPELLS.LIGHTNING_BOLT.id]: 1,
+    'others': 1,
+  };
+
+  on_initialized() {
+    this.active = this.owner.selectedCombatant.hasTalent(SPELLS.ASCENDANCE_TALENT_ELEMENTAL.id) || this.owner.selectedCombatant.hasHands(ITEMS.SMOLDERING_HEART.id);
+  }
+
+  get rawUpdate() {
+    return this.owner.selectedCombatant.getBuffUptime(SPELLS.ASCENDANCE.id) / this.owner.fightDuration;
+  }
+
+  get AscendanceUptime() {
```

Comment:
Two things: it's standard practice to start properties in JS with a lower case. In addition since the module is already called "Ascendance", this property can just be "uptime".

Reply:
Ooop, yeah will change that later.
I forgot to remove `AscendanceUptime` and rename `rawUpdate` to `uptime`.